synapse/rest/media/v1/thumbnail_resource.py

```diff
@@ -384,8 +399,30 @@ async def _select_and_respond_with_thumbnail(
                 file_info.thumbnail.length,
             )
         else:
+            # This might be because:
+            # 1. We can't create thumbnails for the given media (corrupted or
+            #    unsupported file type)
+            # 2. The thumbnailing process never ran or errored out initially
+            #    when the media was first uploaded (these bugs should be
+            #    reported and fixed).
+            # 3. `dynamic_thumbnails` is disabled (see `homeserver.yaml`) and we
+            #    can't generate one. If this was enabled, Synapse would
+            #    go down a different code path.
```

Comment:
I find this confusing. The comment implies we will reach this `else` clause if `dynamic_thumbnails` is disabled, which is not the case.

I would find it clearer if this point was just removed.

Reply:
> The comment implies we will reach this `else` clause if `dynamic_thumbnails` is disabled, which is not the case.

It's definitely possible. That's exactly what was happening in https://github.com/matrix-org/synapse/issues/13039. `dynamic_thumbnails` was disabled and Synapse didn't generate an thumbnails on upload so it got to this point in the code.

I've updated the comment to be a bit more clear about this. Wdyt?

https://github.com/matrix-org/synapse/blob/6a4292574973d40a4c9de3691bf607b692a4d65a/synapse/rest/media/v1/thumbnail_resource.py#L408-L411

Reply:
I really don't understand. Given the assertion at line 325, how do we reach this point in the code if `dynamic_thumbnails` is disabled?

Reply:
oh wait sorry, I really am confused. This point in the code is *only* reachable if `dynamic_thumbnails` is disabled. 

So actually the problem here is that making that explicit for point 3 suggests that it isn't the case for points 1 and 2.

So you could do:

```
          # This might be because `dynamic_thumbnails` is disabled and:
          # 1. We can't create thumbnails for the given media ...
          # 2. The thumbnailing process never ran or errored out ...
          # 3. We can't generate a thumbnail.
```

... but then I don't really understand what point 3 is saying that isn't already covered by points 1 and 2.

Reply:
Works for me 👍 Updated!